File: 162-FindPeakElement/162-FindPeakElement.js
// Last updated: 4/29/2026, 10:33:35 AM
/**
 * @param {number[]} nums
 * @return {number}
 */
var findPeakElement = function(nums) {
    let left = 0;
    let right = nums.length - 1;
    while(left < right) {
        const mid = Math.floor((left + right) / 2);
        console.log(`left: ${left}, mid: ${mid}, right: ${right}`)
        if(nums[mid] < nums[mid + 1]) {
            left = mid + 1;
        } else {
            right = mid;
        }
    }
    return left;
};

File: 162-FindPeakElement/162-FindPeakElement.py
# Last updated: 4/29/2026, 10:33:39 AM
"""
12321 *
12341 *
14123
"""
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        left = 0
        right = len(nums) - 1
        while left < right:
            mid = (left + right) // 2
            print(f"{left}\t{mid}\t{right}")
            if nums[mid] > nums[mid+1] :
                right = mid
            else:
                left = mid + 1
        print(f"<final>: {left}\t{right}")
        return left
                
            
                